### PR TITLE
Fix Due This Month Timezone Boundary Issue

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context
**Type:** Fix  
**Intent:** Correct "Due This Month" and related timezone boundary misclassification by centralizing timezone-aware boundary calculations, threading a per-user timezone offset through server actions and services, capturing client timezone into a cookie via a provider, and updating tests/mocks so CI is stable and month/day/year filters honor user local time.

### 🧭 Reviewer Guide
**Complexity:** Medium

#### Entry Point
Start with `lib/utils.ts` — it introduces core types and helpers (`calculateMonthBoundaries`, `calculateExtendedQueryBoundaries`, `calculateFilterBoundaries`, `calculateDayFilterBoundaries`, `calculateYearFilterBoundaries`, `isTimestampInMonth`) that compute extended query windows and produce UTC filter boundaries used for post-query validation. Then review `lib/timezone.ts` for server-side cookie parsing and `getUserTimezoneOffset`. Next inspect `lib/services/BillService.ts` and `lib/services/TransactionService.ts` to see how services accept `userTimezoneOffset` and apply post-query filtering and updated query ranges. Review `lib/services/ForecastService.ts` for offset propagation and post-projection filtering. Finally, check `components/common/TimezoneProvider.tsx` and `app/layout.tsx` for client-side offset capture, and verify `__mocks__/lib/timezone.ts`, `jest.setup.ts`, and updated tests (e.g., `lib/utils.test.ts` and service/action tests) for CI behavior.

#### Sensitive Areas
- `lib/utils.ts` - boundary math correctness (midnight UTC conversion, month/year wrap-around, leap years) and exported types used across services.  
- `lib/services/BillService.ts` - month filtering logic, extended query ranges, post-query `isTimestampInMonth` filtering, and the new `GetBillsOptions.userTimezoneOffset`.  
- `lib/services/TransactionService.ts` - updated date-range logic, removal of prior in-file boundary helpers, method signatures now accepting `userTimezoneOffset`, and correctness of aggregations/filters.  
- `lib/services/ForecastService.ts` - propagation of `userTimezoneOffset` into per-month projections and post-projection filtering using `isTimestampInMonth`.  
- `lib/timezone.ts` & `components/common/TimezoneProvider.tsx` - cookie name `oar-tz-offset`, parsing/clamping to [-12, 14], units (hours), and client/server parity for offset semantics.  
- Tests & Mocks - `lib/utils.test.ts`, updated service/action tests, `__mocks__/lib/timezone.ts`, and `jest.setup.ts` — ensure mocks/defaults (offset 0) match CI expectations and regression tests cover UTC edge cases, cross-month/year boundaries, and leap years.

### ⚠️ Risk Assessment
- **Breaking Changes:** No breaking changes - public methods gain optional timezone-offset parameters with safe defaults; runtime behavior is preserved when offset is not supplied.  
- **Migrations/State:** No migrations or state changes

### CODE REVIEW

File: lib/utils.test.ts [hunk 1]
```diff
@@ -1,4 +1,11 @@
-import { cn, generateSlug, clampToEndOfMonth } from './utils';
+import {
+  cn,
+  generateSlug,
+  clampToEndOfMonth,
+  calculateMonthBoundaries,
+  calculateFilterBoundaries,
+  isTimestampInMonth,
+} from './utils';
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply lib/utils.test.ts]: @copilot open a new pull request to this branch to apply changes based on [this feedback](https://github.com/sergeyklay/oar/pull/173#discussion_r2659741419) follow .github/instructions/testing.instructions.md instructions
